### ISSUE TITLE
Make it possible to find out which school/LMS instance an LMS-created group is associated with

### ISSUE THREAD

**Issue by judell** [hypothesis/lms#918]
We can query the group table for groups created in the lms authority, but can't know the lms domain.

Says @jeremydean:

> i have no idea what school they're at and just google around or use my own rolodex 
> to figure it out

Adding the lms domain we got from https://lms.hypothes.is/welcome to the group table would solve the problem.

**Comment by klemay** [comment]
## User story:

As a Hypothesis success/edu team member, I want to check on the engagement of our pilot schools (and those not currently in a pilot) with the LMS app. I want to see:

- How many groups were created at the school
- How many annotations are in those groups
- Which groups have a lot of annotations, which groups barely have any, etc

## The current state of things
At present, I can query Metabase for private groups created in the lms.hypothes.is authority. The query returns:
- Group ID
- Group name
- Creation date
- Creator ID
- Creator Name
- Creator Username
- Users in group
- Annotations in group

In order to know which school each group belongs to, I have to Google the `Creator name`, sometimes adding the `Group name` to narrow it down. Often, I'm not able to figure out which school the group creator is from at all. This makes it impossible to follow up on leads.

## The request
Add the LMS domain for LMS groups to Metabase.

**Comment by judell** [comment]
> Add the LMS domain for LMS groups to Metabase.

I would say, rather, as in the original statement:

> Adding the lms domain we got from https://lms.hypothes.is/welcome to the group table 

i.e. to the group table in postgres, so we can see it through the lens of metabase.

**Comment by robertknight** [comment]
I discussed this issue with Lyza and Sean yesterday. This is not purely an operations task as we may need to modify the LMS app to store additional metadata with the groups it creates. It is possible to link a group with an LMS domain given both the h and lms databases but it involves quite a few joins at present:

1. Look up the `user_id` of at least one member in the group in the `user_group` table in h
2. Look up the `provider` associated with that `user_id` (there should be at least one) in the `user_identity` table.
3. Look up the application instance in the `application_instances` lms table, using the `provider` as a `consumer_key` 
4. Finally, get the `lms_url` and `requesters_email` attributes of that the application instance

Step (3) involves a join between the h and lms databases. Metabase doesn't support doing this directly, and the [recommended approach](https://discourse.metabase.com/t/use-data-from-two-databases-in-one-question/337/3) is to combine the relevant information into one data store. At the relatively small scale of our current database, using a Postgres extension (postgres_fdw) to facilitate cross-DB joins may also be possible.

**Comment by judell** [comment]
Is there a downside to remembering the lti domain in the group table?

**Comment by robertknight** [comment]
I'm moving this back to "In Progress" because the original task as explained in the description isn't resolved yet.

The issue title reads: "Make it possible to find out which school/LMS instance an LMS-created group is associated with"

The definition of "done" for this issue is that there is a way to figure out which school/LMS instance an LMS-created group is associated with, and it is documented on this issue.

My understanding is that one important part of the puzzle has been solved, which is combining tables from the lms and h databases into one analytics database, but that doesn't completely resolve the original issue. The second part is for someone (let's say me for the moment, but others are welcome to have a go) who has such a database to try to put into practice [what I described in the comment above](https://github.com/hypothesis/lms/issues/918#issuecomment-527832865) and figure out if it actually works at all. If not, we'll need to do something along the lines of what Jon suggested [in the original description](https://github.com/hypothesis/lms/issues/918#issue-481768842) and add the extra metadata to groups in h somehow before we can close this.

**Comment by robertknight** [comment]
> 1. Look up the user_id of at least one member in the group in the user_group table in h
> 2. Look up the provider associated with that user_id (there should be at least one) in the user_identity table.
> 3. Look up the application instance in the application_instances lms table, using the provider as a consumer_key
> 4. Finally, get the lms_url and requesters_email attributes of that the application instance

I've realized there is a problem with step (2) here since the `provider` value associated with users in the h database is not the consumer key for the LMS install but instead the `tool_consumer_instance_guid` parameter from the LTI launch. This is meant to be a unique identifier for the instance of an LMS. The LMS domain is often used for example, but this choice varies by LMS. Either way, it isn't the consumer key that for some reason I thought it was when I wrote this. We don't persist the `tool_consumer_instance_guid` anywhere in the lms DB, so there is no means to lookup LMS installs associated with a particular user or group. Therefore we'll need to store the LMS context details with the group in a more direct fashion in order to be able to answer this query at all.
